Importing pandas, numpy, matplotlib, seaborn and warning library

In [184]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import warnings
warnings.filterwarnings("ignore")

Importing Dataset

In [185]:
df=pd.read_csv("S:RestaurantData.csv")

Show top 5 record

In [186]:
df.head()

Restaurant ID         Restaurant Name  Country Code              City  \
0        6317637        Le Petit Souffle           162       Makati City   
1        6304287        Izakaya Kikufuji           162       Makati City   
2        6300002  Heat - Edsa Shangri-La           162  Mandaluyong City   
3        6318506                    Ooma           162  Mandaluyong City   
4        6314302             Sambo Kojin           162  Mandaluyong City   

                                             Address  \
0  Third Floor, Century City Mall, Kalayaan Avenu...   
1  Little Tokyo, 2277 Chino Roces Avenue, Legaspi...   
2  Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...   
3  Third Floor, Mega Fashion Hall, SM Megamall, O...   
4  Third Floor, Mega Atrium, SM Megamall, Ortigas...   

                                     Locality  \
0   Century City Mall, Poblacion, Makati City   
1  Little Tokyo, Legaspi Village, Makati City   
2  Edsa Shangri-La, Ortigas, Mandaluyong City   
3      SM Megamall, Ortigas, Mandaluyong City   
4      SM Megamall, Ortigas, Mandaluyong City   

                                    Locality Verbose   Longitude   Latitude  \
0  Century City Mall, Poblacion, Makati City, Mak...  121.027535  14.565443   
1  Little Tokyo, Legaspi Village, Makati City, Ma...  121.014101  14.553708   
2  Edsa Shangri-La, Ortigas, Mandaluyong City, Ma...  121.056831  14.581404   
3  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.056475  14.585318   
4  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.057508  14.584450   

                           Cuisines  ...          Currency Has Table booking  \
0        French, Japanese, Desserts  ...  Botswana Pula(P)               Yes   
1                          Japanese  ...  Botswana Pula(P)               Yes   
2  Seafood, Asian, Filipino, Indian  ...  Botswana Pula(P)               Yes   
3                   Japanese, Sushi  ...  Botswana Pula(P)                No   
4                  Japanese, Korean  ...  Botswana Pula(P)               Yes   

  Has Online delivery Is delivering now Switch to order menu Price range  \
0                  No                No                   No           3   
1                  No                No                   No           3   
2                  No                No                   No           4   
3                  No                No                   No           4   
4                  No                No                   No           4   

   Aggregate rating  Rating color Rating text Votes  
0               4.8    Dark Green   Excellent   314  
1               4.5    Dark Green   Excellent   591  
2               4.4         Green   Very Good   270  
3               4.9    Dark Green   Excellent   365  
4               4.8    Dark Green   Excellent   229  

[5 rows x 21 columns]

shape of the dataset

In [187]:
df.shape

(9551, 21)

Dataset Information
* restaurant id
* restaurant name
* country code
* city
* address
* locality
* locality verbose
* longitude
* latitude
* cuisines
* average cost for two
* currency
* has table booking
* has online delivery
* is delivering now
* switch to order menu
* price range
* ratings
* rating color
* rating text
* votes

we will check data
* check missing values
* check duplicates
* check data type
* check the number of unique values of each column
* check statistics of dataset
* check various categories present in the different categorical column


check missing values

In [188]:
df.isna().sum()

Restaurant ID           0
Restaurant Name         0
Country Code            0
City                    0
Address                 0
Locality                0
Locality Verbose        0
Longitude               0
Latitude                0
Cuisines                9
Average Cost for two    0
Currency                0
Has Table booking       0
Has Online delivery     0
Is delivering now       0
Switch to order menu    0
Price range             0
Aggregate rating        0
Rating color            0
Rating text             0
Votes                   0
dtype: int64

we have 9 null values which are present in cuisines and we will drop that all 9 and it will not impact because we have total 9551 rows


In [189]:
df.dropna(inplace=True)
df.isnull().sum()


Restaurant ID           0
Restaurant Name         0
Country Code            0
City                    0
Address                 0
Locality                0
Locality Verbose        0
Longitude               0
Latitude                0
Cuisines                0
Average Cost for two    0
Currency                0
Has Table booking       0
Has Online delivery     0
Is delivering now       0
Switch to order menu    0
Price range             0
Aggregate rating        0
Rating color            0
Rating text             0
Votes                   0
dtype: int64

after dropping we have zero missing value in the dataset

Check duplicates

In [190]:
df.duplicated().sum()

np.int64(0)

No duplicates value in the dataset


check data types


In [191]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 9542 entries, 0 to 9550
Data columns (total 21 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Restaurant ID         9542 non-null   int64  
 1   Restaurant Name       9542 non-null   object 
 2   Country Code          9542 non-null   int64  
 3   City                  9542 non-null   object 
 4   Address               9542 non-null   object 
 5   Locality              9542 non-null   object 
 6   Locality Verbose      9542 non-null   object 
 7   Longitude             9542 non-null   float64
 8   Latitude              9542 non-null   float64
 9   Cuisines              9542 non-null   object 
 10  Average Cost for two  9542 non-null   int64  
 11  Currency              9542 non-null   object 
 12  Has Table booking     9542 non-null   object 
 13  Has Online delivery   9542 non-null   object 
 14  Is delivering now     9542 non-null   object 
 15  Switch to order menu  9542

Check the number of unique values in each column


In [192]:
df.nunique()

Restaurant ID           9542
Restaurant Name         7437
Country Code              15
City                     140
Address                 8910
Locality                1206
Locality Verbose        1263
Longitude               8111
Latitude                8668
Cuisines                1825
Average Cost for two     140
Currency                  12
Has Table booking          2
Has Online delivery        2
Is delivering now          2
Switch to order menu       1
Price range                4
Aggregate rating          33
Rating color               6
Rating text                6
Votes                   1012
dtype: int64

check statistics of dataset

In [193]:
df.describe()

Restaurant ID  Country Code    Longitude     Latitude  \
count   9.542000e+03   9542.000000  9542.000000  9542.000000   
mean    9.043301e+06     18.179208    64.274997    25.848532   
std     8.791967e+06     56.451600    41.197602    11.010094   
min     5.300000e+01      1.000000  -157.948486   -41.330428   
25%     3.019312e+05      1.000000    77.081565    28.478658   
50%     6.002726e+06      1.000000    77.192031    28.570444   
75%     1.835260e+07      1.000000    77.282043    28.642711   
max     1.850065e+07    216.000000   174.832089    55.976980   

       Average Cost for two  Price range  Aggregate rating         Votes  
count           9542.000000  9542.000000       9542.000000   9542.000000  
mean            1200.326137     1.804968          2.665238    156.772060  
std            16128.743876     0.905563          1.516588    430.203324  
min                0.000000     1.000000          0.000000      0.000000  
25%              250.000000     1.000000          2.500000      5.000000  
50%              400.000000     2.000000          3.200000     31.000000  
75%              700.000000     2.000000          3.700000    130.000000  
max           800000.000000     4.000000          4.900000  10934.000000

# FEATURE ENGINEERING


We will drop this columns
* restaurant id
* restaurant name
* address
* locality verbose
* switch to order menus
This all are identifiers and we are dropping this

In [194]:
columns_to_drop=[
    'Restaurant ID',
    'Restaurant Name',
    'Locality Verbose',
    'Switch to order menu',
    'Address'
]
df.drop(columns=columns_to_drop,inplace=True,errors='ignore')



In [195]:
df.shape

(9542, 16)

ENCODING CATEGORICAL VALUES
# BINARY CATEGORICAL
## Label Encoding
* table booking
* online booking
* delivering now


In [196]:
from sklearn.preprocessing import LabelEncoder
lab=LabelEncoder()
x=df['Has Table booking']
df['Has Table booking']=lab.fit_transform(x)

y=df['Has Online delivery']
df['Has Online delivery']=lab.fit_transform(y)

s=df['Is delivering now']
df['Is delivering now']=lab.fit_transform(s)


# ORDINAL CATEGORICAL
## Ordinal Encoding
* rating text
* rating color

In [197]:
from sklearn.preprocessing import OrdinalEncoder
mab=OrdinalEncoder()
df['Rating text']=mab.fit_transform(df[['Rating text']])

df['Rating color']=mab.fit_transform(df[['Rating color']])


# NOMINAL CATEGORICAL
* country code = One hot encoding
* currency = one hot encoding
* city = frequency encoding
* locality = target encoding

In [203]:
from sklearn.preprocessing import OneHotEncoder,TargetEncoder
hp=OneHotEncoder()
ph=TargetEncoder()

df['Country Code']=hp.fit_transform(df['Country Code'])
df['Currency']=hp.fit_transform(df['Currency'])

df['locality']=ph.fit_transform(df['locality'])


ValueError: Expected a 2-dimensional container but got <class 'pandas.core.series.Series'> instead. Pass a DataFrame containing a single row (i.e. single sample) or a single column (i.e. single feature) instead.

# Multi label feature
* cuisines = multi label encoding

In [204]:
from sklearn.preprocessing import MultiLabelBinarizer
bs=MultiLabelBinarizer()
df['Cuisines']=bs.fit_transform(df['Cuisines'])
